plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialLeaves.java

```diff
@@ -0,0 +1,108 @@
+package com.denizenscript.denizen.objects.properties.material;
+
+import com.denizenscript.denizen.objects.MaterialTag;
+import com.denizenscript.denizen.utilities.debugging.Debug;
+import com.denizenscript.denizencore.objects.Mechanism;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.properties.Property;
+import com.denizenscript.denizencore.tags.Attribute;
+import org.bukkit.block.data.Bisected;
+import org.bukkit.block.data.type.Leaves;
+
+public class MaterialLeaves implements Property {
+
+    public static boolean describes(ObjectTag material) {
+        return material instanceof MaterialTag
+                && ((MaterialTag) material).hasModernData()
+                && ((MaterialTag) material).getModernData().data instanceof Leaves;
+    }
+
+    public static MaterialLeaves getFrom(ObjectTag _material) {
+        if (!describes(_material)) {
+            return null;
+        }
+        else {
+            return new MaterialLeaves((MaterialTag) _material);
+        }
+    }
+
+    public static final String[] handledTags = new String[] {
+            "persistent", "tree_distance"
+    };
+
+    public static final String[] handledMechs = new String[] {
+            "persistent"
+    };
+
+
+    private MaterialLeaves(MaterialTag _material) {
+        material = _material;
+    }
+
+    MaterialTag material;
+
+    @Override
+    public String getAttribute(Attribute attribute) {
+
+        if (attribute == null) {
+            return null;
+        }
+
+        // <--[tag]
+        // @attribute <MaterialTag.persistent>
+        // @returns ElementTag(Boolean)
+        // @group properties
+        // @description
+        // Returns if this block will decay from being too far away from a tree.
+        // -->
+        if (attribute.startsWith("persistent")) {
+            return new ElementTag(getLeaves().isPersistent()).getAttribute(attribute.fulfill(1));
+        }
+
+        // <--[tag]
+        // @attribute <MaterialTag.tree_distance>
+        // @returns ElementTag(Number)
+        // @group properties
+        // @description
+        // Returns a number of how many blocks away from a connected tree it is.
+        // Defaults to 7 if not connected to a tree.

```

Comment:
This... seems like it should be a property of its own? If it's measured by the material itself...

Reply:
I added the tag, as it's part of the leaves interface. It does have a getter and setter, but I couldnt see any implementation where it'd be useful to use the setter over just using persistence.